asf_heat_pump_affordability/notebooks/investigate_effect_of_room_number.py

```diff
@@ -0,0 +1,496 @@
+#       extension: .py
+#       format_name: light
+#       format_version: '1.5'
+#       jupytext_version: 1.16.1
+#   kernelspec:
+#     display_name: Python 3 (ipykernel)
+#     language: python
+#     name: python3
+# ---
+
+# ## Question: Does pre/post 1950 explain differences in ASHP installation costs within archetypes?
+#
+# Or does property size (i.e. number of rooms) explain the difference in cost within archetypes?
+#
+# To investigate this, we will:
+# 1. Preprocess the MCS-EPC dataset to get adjusted costs of ASHP installation against a base year of 2023. This dataset will be used in the subsequent analyses.
+# 2. Look at the distribution of number of rooms across the pre- and post-1950 groupings of the different 4 property types and see whether there are differences.
+# 3. Create 2 series of quantile regression models to compare how predictions of installation costs change when number of rooms is included as a feature or not.
+
+# +
+from io import StringIO
+import regex as re
+import numpy as np
+import math
+import pandas as pd
+import matplotlib.pyplot as plt
+import statsmodels.formula.api as sm
+
+from asf_heat_pump_affordability import config
+from asf_heat_pump_affordability.getters import get_data
+from asf_heat_pump_affordability.pipeline import (
+    preprocess_cpi,
+    preprocess_data,
+    archetypes,
+)
+
+# -
+
+# ## 1. Get and preprocess dataset
+
+## Get MCS-EPC data
+data = pd.read_csv(
+    "s3://asf-heat-pump-affordability/mcs_installations_epc_most_relevant_231009_preprocessed_yearRange_20212023.csv",
+    index_col=0,
+)
+
+# +
+## Get CPI data with adjustment factors for 2023 base year
+cpi_05_3_df = get_data.get_df_from_url(config["data_source"]["cpi_source_url"])
+cpi_quarterly_df = preprocess_cpi.get_df_quarterly_cpi_with_adjustment_factors(
+    ref_year=2023,
+    cpi_df=cpi_05_3_df,
+    cpi_col_header=config["cpi_data"]["cpi_column_header"],
+)
+
+## Get MCS-EPC df with adjusted costs
+data = preprocess_data.generate_df_adjusted_costs(
+    mcs_epc_df=data, cpi_quarters_df=cpi_quarterly_df
+)
+
+# +
+## Add archetypes
+
+for (
+    archetype,
+    filter,
+) in archetypes.classify_dict_archetypes_masks(data).items():
+    data.loc[filter, "archetype"] = archetype
+# -
+
+data.groupby("archetype")["archetype"].count()
+
+# +
+## As we will use number of habitable rooms as a feature, we need to drop rows with NaN values or 0 values
+## View count of lost values from each archetype
+
+data[
+    (data["NUMBER_HABITABLE_ROOMS"].isna()) | (data["NUMBER_HABITABLE_ROOMS"] == 0)
+].groupby("archetype")["archetype"].count().sort_values(ascending=False)
+
+# +
+_data = data.copy()
+
+# Remove park homes as low count
+_data = _data.loc[_data["PROPERTY_TYPE"] != "Park home"]
+
+# Remove rows with 0 or NaN number of rooms
+_data = _data[_data["NUMBER_HABITABLE_ROOMS"] != 0]
+_data.dropna(subset=["NUMBER_HABITABLE_ROOMS"], inplace=True)
+
+
+# Create archetype4 column (i.e. property type)
```

Comment:
You can make this simpler by doing:

`_data["archetype4"] = _data["archetype"].str.split("_", n=2, expand=True)[2]`

Reply:
wow never knew you could do such customisable stuff with split!